virtcontainers/kata_agent.go

```diff
@@ -777,16 +776,43 @@ func (k *kataAgent) createContainer(sandbox *Sandbox, c *Container) (p *Process,
 	// We need to give the OCI spec our absolute rootfs path in the guest.
 	grpcSpec.Root.Path = rootPath
 
+	sharedPidNs := false
+	pidNsIndex := -1
+
+	// Sandbox pid will be set once the sandbox container is created.
+	// Check if pid namespace path is the same as the sandbox/infra container
+	if sandbox.state.Pid > 0 {
+		for i, ns := range grpcSpec.Linux.Namespaces {
+			if ns.Type == string(specs.PIDNamespace) {
+				if ns.Path == fmt.Sprintf("/proc/%d/ns/pid", sandbox.state.Pid) {
```

Comment:
Are we sure this is the only form a namespace path could have ? What if a symlink is provided ? Is it possible to find the root namespace to make sure we don't miss some corner cases ?

Reply:
yeah, I'll check for symlinks too